tests/kernel/interrupt/src/static_isr.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <kernel_structs.h>
+#include "interrupt_util.h"
+#include <kernel_internal.h>
+
+#define ISR0_ARG        0xca55e77e
+#define ISR1_ARG	0xb01dface
+#define DURATION	5
+#define TEST_IRQ_PRI	1
+
+/*
+ * We try to choose the unused irq that fitting identical category of platform
+ * as possible.
+ */
+#if defined(CONFIG_CPU_CORTEX_M)
+
+#if CONFIG_NUM_IRQS >= 96
+#define TEST_IRQ_0_LINE (CONFIG_NUM_IRQS - 3)
+#define TEST_IRQ_1_LINE (CONFIG_NUM_IRQS - 4)
+#elif CONFIG_NUM_IRQS <= 8
+#define TEST_IRQ_0_LINE 5
+#define TEST_IRQ_1_LINE 6
+#elif CONFIG_NUM_IRQS > 8 && CONFIG_NUM_IRQS < 32
+#define TEST_IRQ_0_LINE 18
+#define TEST_IRQ_1_LINE 19
+#else
+#define TEST_IRQ_0_LINE 22
+#define TEST_IRQ_1_LINE 29
+#endif
+
+#elif defined(CONFIG_GIC)
+
+#define TEST_IRQ_0_LINE 14
+#define TEST_IRQ_1_LINE 15
+
+#elif defined(CONFIG_CPU_ARCV2)
+/*
+ * Use 2th and 3th from last is because they won't be multiple registrations in
+ * all ARC platform.
+ */
+#define TEST_IRQ_0_LINE (CONFIG_NUM_IRQS - 2)
+#define TEST_IRQ_1_LINE (CONFIG_NUM_IRQS - 3)
+
+#elif defined(CONFIG_X86)
+
+#define TEST_IRQ_0_LINE 16
+#define TEST_IRQ_1_LINE 17
+
+#elif defined(CONFIG_ARCH_POSIX)
+
+#define TEST_IRQ_0_LINE 6
+#define TEST_IRQ_1_LINE 7
+
+#elif defined(CONFIG_RISCV)
+
+#if defined(CONFIG_SOC_RISCV32_LITEX_VEXRISCV)
+/* For Litex vexriscv board, irq 1 has been used */
+#define TEST_IRQ_0_LINE 2
+#else
+#define TEST_IRQ_0_LINE 1
+#endif
+
+#define TEST_IRQ_1_LINE 5
+
+#elif defined(CONFIG_XTENSA)
+
+#define TEST_IRQ_0_LINE 8
+#define TEST_IRQ_1_LINE 9
+
+#else
+#endif
+
+/* Helper function for getting current stack pointer */
+static inline void get_stack_addr(uintptr_t *reg)
```

Comment:
Too complicated.  You don't care about precision here, you just want "any address on the local stack".  So do it portably and take the address of a local variable:

```c
static inline void *get_stack_addr(uintptr_t *reg)
{
    int dummy;
    *reg = &dummy;
}
```

Reply:
You are right, This is a good arch-neutral way to do that, learn a lot again, thanks!